### PR TITLE
fix bug where mobs says in a non git directory for mob version you need to be in a git repository

### LINKED ISSUE #380: BUG: Mob v4.4.1 won't display config, help, or version when the current directory is not a git repo
Upgraded to mob v4.4.1 on macOS 12 (via brew).

After updating, the `config`, `help`, `moo`, and `version` commands and switches return an error when the directory is not a git repo. This is a change from v4.4.0.

The following commands return an error message when the directory is not a git repo.

```sh
mob config
mob help
mob moo
mob version
mob --help
mob --version
```

The error message is:

```sh
ERROR expecting the current working directory to be a git repository
```

These commands do not operate on a git repo and should not require a git repo.

(I expect there is a change between 4.4.0 and 4.4.1 that (incorrectly) moved a test for a git repo to be inclusive of all commands.)

### PR DESCRIPTION
**Pull request by hollesse**
closes #380

### CODE REVIEW

**Comment by hollesse** [comment]
I refactored the tests and extracted a function run out of the main function to better test the code. So the integration tests where i was really testing the executable are not needed anymore.

File: mob_test.go
Comment (gregorriegler) [inline mob_test.go:1]: What happens if you just run the tests but you don't have built the mob binary beforehand?

Comment (hollesse) [reply mob_test.go:1]: Just found out my push did not worked. I refactored the tests such that they are not integrationtests anymore

**Review by gregorriegler — APPROVED** [review]
nice!

**Pull request merged**